lib/backend-operations/ops-cloud-api.js

```diff
@@ -209,45 +193,19 @@ function isCompleteLamdbaFunction(resource){
                             resource.attributes.status.includes('COMPLETE'))
 }
 
-//////////////////// for build ////////////////////
-
-function isNewBuildNeeded(){
-    let result = false
-    if(fs.existsSync(_featureDirPath)){
-        if(fs.existsSync(_featureBuildDirPath)){
-            let lastBackendBuildTime = moment(_projectInfo.BackendLastBuildTime, awsmobileJSConstant.DateTimeFormatString)
-            let lastBackendSpecModificationTime = moment(backendSpecManager.getLastModificationTime(_projectInfo))
-            let lastFeatureModificationTime = moment(dfops.getDirContentMTime(_featureDirPath))
-            let lastFeatureBuildModificationTime = moment(dfops.getDirContentMTime(_featureBuildDirPath))
-
-            result =!lastBackendBuildTime.isValid() ||
-                    !lastBackendSpecModificationTime.isValid() ||
-                    !lastFeatureModificationTime.isValid() ||
-                    !lastFeatureBuildModificationTime.isValid() ||
-                    lastBackendBuildTime.isBefore(lastBackendSpecModificationTime) ||
-                    lastBackendBuildTime.isBefore(lastFeatureModificationTime) ||
-                    lastBackendBuildTime.isBefore(lastFeatureBuildModificationTime) //user manually changed build
-        }else{
-            result = true
-        }
-    }else{ // the user has removed the feature dir
-        fs.ensureDirSync(_featureDirPath)
-        result = false
-    }
-    
-    return result
-}
-
 /////////////////Cloud Formation Logic/////////////////
 function getStateGroup(stateSummary){
-    //-1: unrecognized
+    //-2: unrecognized
+    //-1: not-yet-deployed
     // 0: in-progress
     // 1: terminal_complate
     // 2: terminal_failed
-    let result = -1 
+    let result = -2 
     if(cloudFormationCodes.stackStatusCodes.includes(stateSummary)){
         result = 0
-        if(cloudFormationCodes.terminalState_Complete.includes(stateSummary)){
+        if(cloudFormationCodes.terminalState_NotDeployed.includes(stateSummary)){
```

Comment:
Why includes and not equals?

Reply:
Just in case they add more states that do not belong to other categories, so we can just add it to that category